hack/test-integration.sh

```diff
@@ -99,19 +105,30 @@ trap cleanup EXIT SIGINT
 function exectest() {
 	echo "Running $1..."
 
-	result=1
-	if [ -n "${VERBOSE-}" ]; then
-		ETCD_PORT=${ETCD_PORT} "${testexec}" -test.v -test.run="^$1$" "${@:2}" 2>&1
-		result=$?
-	else
-		out=$(ETCD_PORT=${ETCD_PORT} "${testexec}" -test.run="^$1$" "${@:2}" 2>&1)
-		result=$?
+	local skipped=0
+	local result=1
+	out=$(ETCD_PORT=${ETCD_PORT} "${testexec}" -test.v -test.run="^$1$" "${@:2}" 2>&1)
+	result=$?
+	if echo "$out" | grep -q '^---\s\+SKIP'; then
+		skipped=1
+	fi
+	if [[ -n "${VERBOSE:-}" ]]; then
+		echo "$out"
 	fi
 
 	tput cuu 1 # Move up one line
 	tput el		# Clear "running" line
 
-	if [[ ${result} -eq 0 ]]; then
+	if [[ $skipped = 1 ]]; then
```

Comment:
move the skip check inside the if ${result} -eq 0... skips should qualify a 0 return code that would otherwise be reported as success, not excuse a non-zero return code